### PR TITLE
Coarsen timestamp value

### PR DESCRIPTION
**Pull request by beaufortfrancois**
This PR fixes https://github.com/gpuweb/gpuweb/issues/4175

@kainino0x Please review.

### CODE REVIEW

File: spec/index.bs [hunk 1]
```diff
@@ -13129,8 +13128,14 @@ be rare and can safely be ignored. Applications should not be written in such a
 timestamps cause an application failure.
 
 <p tracking-vector>
-Timestamp queries can provide high-resolution GPU timing.
-See [[#security-timing]] for security considerations.
+Timestamp queries can provide high-resolution GPU timing. To mitigate privacy concerns, they must be
+processed with the following algorithm:
+
+<div algorithm>
+    <dfn abstract-op>Process timestamp value</dfn>
```
Comment (kainino0x) [inline spec/index.bs:13135]: We need to reference this from the pseudocode which implements `writeTimestamp` and `GPU{Render,Compute}TimestampWrites.{beginning,end}OfPassWriteIndex`.

I think we should write this algorithm as `To get the <dfn abstract-op>current queue timestamp</dfn>:` and then replace the places where it currently says "current timestamp for the Queue timeline" or "current queue timestamp".

Comment (beaufortfrancois) [reply spec/index.bs]: Done

File: spec/index.bs [hunk 2]
```diff
@@ -13129,8 +13128,14 @@ be rare and can safely be ignored. Applications should not be written in such a
...
+
+    Return the result of calling [=coarsen time=] with the timestamp value.
```
[the remark this answers is not in the document]
Comment (beaufortfrancois) [reply spec/index.bs:13137]: Done

**Comment by beaufortfrancois** [comment]
@kainino0x Please have another look

File: spec/index.bs [hunk 3]
```diff
@@ -589,10 +589,9 @@ serialize the [=agents=] sharing any handles to prevent any concurrency entirely
 In the end, the attack surface for races on shared state in WebGPU will be
 a small subset of the `SharedArrayBuffer` attacks.
 
-WebGPU also specifies the {{GPUFeatureName/"timestamp-query"}} feature, which
-provides high precision timing of GPU operations. The feature is optional, and a WebGPU
-implementation may limit its exposure only to those scenarios that are trusted. Alternatively,
-the timing query results could be processed by a compute shader and aligned to a lower precision.
+WebGPU also specifies the optional {{GPUFeatureName/"timestamp-query"}} feature, which
+provides high precision timing of GPU operations. To mitigate privacy concerns, the timing query
+values are aligned to a lower precision.
```
Comment (kainino0x) [inline spec/index.bs:594]:
```suggestion
values are aligned to a lower precision: see [$current queue timestamp$].
```

File: spec/index.bs [hunk 4]
```diff
@@ -10048,8 +10047,8 @@ dictionary GPUCommandEncoderDescriptor
             <div data-timeline=queue>
                 [=Queue timeline=] steps:
 
-                1. Write the current timestamp for the [=Queue timeline=], in nanoseconds, into
-                    |querySet| at index |queryIndex|.
+                1. Write the [$current queue timestamp$] for the [=Queue timeline=],
+                    in nanoseconds, into |querySet| at index |queryIndex|.
```
Comment (kainino0x) [inline spec/index.bs:10051]:
```suggestion
                1. Write the [$current queue timestamp$] into |querySet| at index |queryIndex|.
```

File: spec/index.bs [hunk 5]
```diff
@@ -13129,8 +13128,17 @@ be rare and can safely be ignored. Applications should not be written in such a
...
 <p tracking-vector>
-Timestamp queries can provide high-resolution GPU timing.
-See [[#security-timing]] for security considerations.
+Timestamp queries can provide high-resolution GPU timing. To mitigate privacy concerns, they are
+aligned to a lower precision with the following algorithm:
+
+<div algorithm>
+    To get the <dfn abstract-op>current queue timestamp</dfn>:
+    
+    Return the result of calling [=coarsen time=] with the timestamp value.
```
Comment (kainino0x) [inline spec/index.bs:13137]:
```suggestion
    - Let |fineTimestamp| be the current timestamp value of the current [=queue timeline=],
        in nanoseconds, relative to an implementation-defined point in the past.
    - Return the result of calling [=coarsen time=] on |fineTimestamp|.
```

The "implementation-defined point in the past" thing is intentionally underspecified - it isn't guaranteed to always be the *same* point in the past - filed #4361.

**Review by kainino0x — APPROVED** [review]

File: spec/index.bs [hunk 6]
```diff
@@ -589,10 +589,9 @@ serialize the [=agents=] sharing any handles to prevent any concurrency entirely
 In the end, the attack surface for races on shared state in WebGPU will be
 a small subset of the `SharedArrayBuffer` attacks.
 
-WebGPU also specifies the {{GPUFeatureName/"timestamp-query"}} feature, which
-provides high precision timing of GPU operations. The feature is optional, and a WebGPU
-implementation may limit its exposure only to those scenarios that are trusted. Alternatively,
-the timing query results could be processed by a compute shader and aligned to a lower precision.
+WebGPU also specifies the optional {{GPUFeatureName/"timestamp-query"}} feature, which
+provides high precision timing of GPU operations. To mitigate privacy concerns, the timing query
```
Comment (kainino0x) [inline spec/index.bs:593]:
```suggestion
provides high precision timing of GPU operations. To mitigate security and privacy concerns, the timing query
```

File: spec/index.bs [hunk 7]
```diff
@@ -13129,8 +13127,19 @@ be rare and can safely be ignored. Applications should not be written in such a
 timestamps cause an application failure.
 
 <p tracking-vector>
-Timestamp queries can provide high-resolution GPU timing.
-See [[#security-timing]] for security considerations.
+Timestamp queries can provide high-resolution GPU timing. To mitigate privacy concerns, they are
+aligned to a lower precision with the following algorithm:
```
Comment (kainino0x) [inline spec/index.bs:13131]:
```suggestion
Timestamp queries could provide high-resolution GPU timing (see [[#security-timing]]).
To mitigate security and privacy concerns, their precision is reduced in the following algorithm:
```

**Pull request merged**